docs/docs/recipes/protect-your-api/fragments/_token_validation.mdx

```diff
@@ -128,5 +128,56 @@ Package selections vary depending on the framework you are using.
 I.e., checkout [org.springframework.security.oauth2.jwt](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/oauth2/jwt/package-summary.html), if you use Spring Framework.
 :::
 
+</TabItem>
+<TabItem value="python" label="Python">
+
+Leverage authlib's [JWTBearerTokenValidator](https://github.com/lepture/authlib/blob/169c7dcfc47478c8d55553cc95fb0f5578162b77/authlib/oauth2/rfc7523/validator.py)
+
+```python
+from authlib.oauth2.rfc7523 import JWTBearerTokenValidator
+from authlib.jose.rfc7517.jwk import JsonWebKey
+
+
+class MyLogtoBearerTokenValidator(JWTBearerTokenValidator):
+    """
+    jwks: You will need to inquire the jwks from Logto's jwks endpoint: "https://<your-logto-domain>/oidc/jwks" and pass in the response
+    """
+    def __init__(self, jwks, issuer="https://<your-logto-domain>/oidc", audience):
+        public_key = JsonWebKey.import_key_set(jwks)
+        super(MyLogtoBearerTokenValidator, self).__init__(public_key, issuer)
+        self.claims_options = {
+           'exp': {'essential': True},
+           'iss': {'essential': True, 'value': issuer}
+           'aud': {"essential": True, "value": audience}
+        }
+
+```
+
+To guard the token audience more strictly at each endpoint level:
+
+```python
+from authlib.oauth2.rfc7523 import JWTBearerTokenValidator
+from authlib.jose.rfc7517.jwk import JsonWebKey
+from authlib.oauth2.rfc6749.errors
+
+
+class MyLogtoBearerTokenValidator(JWTBearerTokenValidator):
+    """
+    jwks: You will need to inquire the jwks from Logto's jwks endpoint: "https://<your-logto-domain>/oidc/jwks" and pass in the response
+    """
+    def __init__(self, jwks, issuer="https://<your-logto-domain>/oidc"):
+        public_key = JsonWebKey.import_key_set(jwks)
+        super(MyLogtoBearerTokenValidator, self).__init__(public_key, issuer)
+        self.claims_options = {
+           'exp': {'essential': True},
+           'iss': {'essential': True, 'value': issuer}
+        }
+    def validate_token(self, token, scopes, request):
+        super(MyLogtoBearerTokenValidator, self).validate_token(token, scopes, request)
+        if token.get('aud') != str(request.url)
```

Comment:
I think we can combine these two examples, and mark this function as optional? should be neater

Reply:
updated